src/DocumentDB.ChangeFeedProcessor/LeaseManagement/PartitionedByPartitionKeyCollectionRequestOptionsFactory.cs

```diff
@@ -0,0 +1,48 @@
+﻿//----------------------------------------------------------------
+// Copyright (c) Microsoft Corporation.  Licensed under the MIT license.
+//----------------------------------------------------------------
+
+namespace Microsoft.Azure.Documents.ChangeFeedProcessor.LeaseManagement
+{
+    using System;
+    using Microsoft.Azure.Documents.ChangeFeedProcessor.PartitionManagement;
+    using Microsoft.Azure.Documents.Client;
+
+    /// <summary>
+    /// Used to create request options for partitioned lease collections, when partition key is defined as /partitionKey.
+    /// </summary>
+    internal class PartitionedByPartitionKeyCollectionRequestOptionsFactory : IRequestOptionsFactory
+    {
+        public RequestOptions CreateRequestOptions(ILease lease)
+        {
+            var documentServiceLease = lease as DocumentServiceLease;
+            if (documentServiceLease == null)
+                throw new ArgumentException("lease is not of type DocumentServiceLease");
```

Comment:
nit: add parameter name to the list of args, i.e. nameof(lease), for the other method as well.